packages/diagram/src/diagramEngine.ts

```diff
@@ -99,8 +99,15 @@ export class DiagramEngine {
         const layoutErrors: Error[] = [];
         const interpretationResult = this.execute(expressions, config);
         if (interpretationResult.result != undefined) {
-            assertWrapperObject(interpretationResult.result, "Returned diagram");
-            layoutedDiagram = await this.layoutEngine.layout(interpretationResult.result.wrapped, config);
+            try {
+                const diagram = interpretationResult.result;
+                if (!isWrapperObject(diagram) || !(diagram.wrapped instanceof LayoutWithRoot)) {
+                    throw new RuntimeError("No diagram returned");
+                }
+                layoutedDiagram = await this.layoutEngine.layout(diagram.wrapped, config);
```

Comment:
Can `layout` throw?
If no, you can inline below into 105

Reply:
yes
e.g. during the circular relative point detection